aws-iot-scheduledaudit/src/main/java/com/amazonaws/iot/scheduledaudit/CreateHandler.java

```diff
@@ -0,0 +1,91 @@
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+import java.util.Map;
+
+public class CreateHandler extends BaseHandler<CallbackContext> {
+
+    // Copied value from software.amazon.cloudformation.resource.IdentifierUtils
+    private static final int GENERATED_NAME_MAX_LENGTH = 40;
+
+    private final IotClient iotClient;
+
+    public CreateHandler() {
+        iotClient = IotClient.builder().build();
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        CreateScheduledAuditRequest createRequest = translateToCreateRequest(request);
+
+        ResourceModel model = request.getDesiredResourceState();
+        if (!StringUtils.isEmpty(model.getScheduledAuditArn())) {
+            logger.log(String.format("ScheduledAuditArn is read-only, but the caller passed %s.",
+                    model.getScheduledAuditArn()));
+            // Note: this is necessary even though ScheduledAuditArn is marked readOnly in the schema.
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    "ScheduledAuditArn is a read-only property and cannot be set.");
+        }
+
+        CreateScheduledAuditResponse createScheduledAuditResponse;
+        try {
+            createScheduledAuditResponse = proxy.injectCredentialsAndInvokeV2(
+                    createRequest, iotClient::createScheduledAudit);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+
+        model.setScheduledAuditArn(createScheduledAuditResponse.scheduledAuditArn());
+        logger.log(String.format("Created %s.", createScheduledAuditResponse.scheduledAuditArn()));
+
+        return ProgressEvent.defaultSuccessHandler(model);
+    }
+
+    private CreateScheduledAuditRequest translateToCreateRequest(ResourceHandlerRequest<ResourceModel> request) {
+
+        ResourceModel model = request.getDesiredResourceState();
+
+        // Like most services, we don't require an explicit resource name in the template,
+        // and, if it's not provided, generate one based on the stack ID and logical ID.
+        if (StringUtils.isBlank(model.getScheduledAuditName())) {
+            model.setScheduledAuditName(IdentifierUtils.generateResourceIdentifier(
+                    request.getStackId(), request.getLogicalResourceIdentifier(),
```

Comment:
StackId seems to be an ARN. Usually the practice I have seen is to use the stack name to generate the resource name, is that what we are doing? Does this class extracts stack name from stackId/arn?

Reply:
Yes, `IdentifierUtils.generateResourceIdentifier` extracts the stack name from the stackId.